### ISSUE TITLE
implement thread safety?

### ISSUE THREAD

**Issue by monomycelium** [phip1611/generic-tm1637-gpio-driver-rust#6]
Thank you for creating this great library! I've found it helpful whenever I want to write a small app to use the TM1637 display. Recently, I've been working on a project to control the display interactively from the internet. I managed to get most things working using the [`warp`](https://crates.io/crates/warp) crate. However, I could not access the `TM1637Adapter` from the serving thread. The `TM1637Adapter` seemed to have to implement the `Send` and `Sync` traits. Even using other wrappers like `std::sync::Arc` and `std::sync::Mutex`, I could not pass a reference to the adapter to the serving thread. Is it possible for `TM1637Adapter` to be thread-safe, or do I need to look into other alternatives like `std::sync::mpsc`?

**Comment by monomycelium** [comment]
After playing around, I found out that only `(dyn Fn() + 'static)` cannot be sent between threads safely. I've seen the other issue, #2, about removing it or making it optional, and removing it would make thread safety possible. I hope you would consider doing so.

**Comment by monomycelium** [comment]
Before the Rust server, I used [`wsServer`](https://github.com/Theldus/wsServer) in C and wrapped functions for your crate by exporting them using Rust's FFI. That was working like a charm, with multithreaded connections, yet the ability to write to the display safely from each thread, but I had to switch only because of the lack of SSL support in `wsServer`. I guess I shall move on to using either `std::sync::mpsc` or the greatest, C.
